src/app/components/merge/ContactDetails.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import { useApi, Loader, FormModal, PrimaryButton } from 'react-components';
+
+import { getContact } from 'proton-shared/lib/api/contacts';
+import { prepareContact, bothUserKeys } from '../../helpers/decrypt';
+
+import ContactViewErrors from '../ContactViewErrors';
+import ExtendedContactSummary from './ExtendedContactSummary';
+
+import { FAIL_TO_LOAD } from '../../constants';
+
+const ContactDetails = ({ contactID, userKeysList, ...rest }) => {
+    const api = useApi();
+    const [loading, setLoading] = useState(true);
```

Comment:
Use `useLoading` please and wrap async process with `withLoading`.

Reply:
Ok! I didn't know about that hook!